### PR TITLE
Move key add, list, remove and passwd as separate sub-commands and improve key sub-command documentation

### LINKED ISSUE #4676: `key` help does not document subcommands
Output of `restic version`
--------------------------
restic 0.16.3 compiled with go1.21.6 on linux/amd64

Problem description
----------------------------------------

`restic help key` shows the list/add/remove/passwd subcommands but does not show help for what they do. Specifically, the difference between "add" and "passwd" is not explained.

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------

Today and every day since I first found it.

### PR DESCRIPTION
**Pull request by konidev20**
What does this PR change? What problem does it solve?
-----------------------------------------------------
Restic previously used command line args to figure out the command to be executed for the `key` command. The documentation of the key command was limited. 

@MichaelEischer requested the key command to be split into their own sub-commands like how the `repair` command is structured.

This PR has split the `key` command into their respective sub-commands using spf13/cobra's sub-command feature. 

Was the change previously discussed in an issue or on the forum?
----------------------------------------------------------------
Closes #4676 

> We should split the key command into multiple subcommands, similiar to the repair command.
> 
> "add" create a new key. "passwd" replaces the current key with a new one using the new password.

Checklist
---------

- [x] I have read the [contribution guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches).
- [x] I have [enabled maintainer edits](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I have added tests for all code changes.
- - reused existing integration tests, fixed the test and checked if everything is working 
- [x] I have added documentation for relevant changes (in the manual).
- [x] There's a new file in `changelog/unreleased/` that describes the changes for our users (see [template](https://github.com/restic/restic/blob/master/changelog/TEMPLATE)).
- [x] I have run `gofmt` on the code in all commits.
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits).
- [x] I'm done! This pull request is ready for review.

### CODE REVIEW

File: cmd/restic/cmd_key_add.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+Exit status is 0 if the command is successful, and non-zero if there was any error.
+	`,
+	DisableAutoGenTag: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return runKeyAdd(cmd.Context(), globalOptions, addKeyOpts)
+	},
+}
+
+type NewPasswordOptions struct {
```
Comment (MichaelEischer) [inline cmd/restic/cmd_key_add.go:30]: There is no longer a reason to keep this as a separate struct. Please inline it into `AddKeyOptions`.

File: cmd/restic/cmd_key_add.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+func init() {
+	cmdKey.AddCommand(cmdKeyAdd)
+
+	flags := cmdKeyAdd.Flags()
+	flags.StringVarP(&addKeyOpts.NewPasswordFile, "new-password-file", "", "", "`file` from which to read the new password")
+	flags.StringVarP(&addKeyOpts.Username, "user", "", "", "the username for new key")
+	flags.StringVarP(&addKeyOpts.Hostname, "host", "", "", "the hostname for new key")
+}
+
+func runKeyAdd(ctx context.Context, gopts GlobalOptions, opts AddKeyOptions) error {
```
Comment (MichaelEischer) [inline cmd/restic/cmd_key_add.go:51]: now that we can do so in a clean way, please return an error if any `args` are passed to `key add`. Please also make the corresponding change to the other commands.

File: cmd/restic/cmd_key_remove.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+		return err
+	}
+
+	lock, ctx, err := lockRepo(ctx, repo, gopts.RetryLock, gopts.JSON)
+	defer unlockRepo(lock)
+	if err != nil {
+		return err
+	}
+
+	id, err := restic.Find(ctx, repo, restic.KeyFile, args[0])
```
Comment (MichaelEischer) [inline cmd/restic/cmd_key_remove.go:46]: The `restic.Find` call should be part of `deleteKey`.

Comment (MichaelEischer) [inline cmd/restic/cmd_key_remove.go:46]: `len(args)` must equal `1` otherwise this could crash.

**Review by MichaelEischer — CHANGES_REQUESTED** [review]
Thanks for working on this! I like the change in general, but I found a few problems that were introduced while refactoring the code.

File: cmd/restic/cmd_key_integration_test.go [hunk 4]
```diff
@@ -130,15 +130,50 @@ func TestKeyProblems(t *testing.T) {
...
+	rtest.Assert(t, err != nil, "expected key list to fail")
+
+	err = runKeyRemove(context.TODO(), env.gopts, []string{})
+	t.Log(err)
+	rtest.Assert(t, err != nil, "expected key remove to fail")
+
+	err = runKeyRemove(context.TODO(), env.gopts, []string{"john", "doe"})
+	t.Log(err)
+	rtest.Assert(t, err != nil, "expected key remove to fail")
+}
```
Comment (konidev20) [inline cmd/restic/cmd_key_integration_test.go:179]: @MichaelEischer I've added this new integration test to check for the invalid parameters for the sub-commands.

**Review by MichaelEischer — APPROVED** [review]
LGTM.

I've fixed the integration test. `key list` returned an error for the wrong reason.

**Pull request merged**